couchdb/tests/server/compiler.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+#
+import binascii
+import types
+import unittest
+
+from couchdb import util
+from couchdb.server import compiler
+from couchdb.server import exceptions
+
+
+class DDocModulesTestCase(unittest.TestCase):
+
+    def test_resolve_module(self):
+        module = {'foo': {'bar': {'baz': '42'}}}
+        mod_info = compiler.resolve_module('foo/bar/baz'.split('/'), {}, module)
+        self.assertEqual(
+            mod_info,
+            {
+                'current': '42',
+                'parent': {
+                    'current': module['foo']['bar'],
+                    'id': 'foo/bar',
+                    'parent': {
+                        'id': 'foo',
+                        'current': module['foo'],
+                        'parent': {'current': module}
+                    },
+                },
+                'id': 'foo/bar/baz',
+                'exports': {},
+            }
+        )
+
+    def test_relative_path(self):
+        module = {'foo': {'bar': {'baz': '42'}}}
+        mod_info = compiler.resolve_module(
+            'foo/./bar/../bar/././././baz/../../bar/baz'.split('/'), {}, module)
+        self.assertTrue(mod_info['id'], 'foo/bar/baz')
```

Comment:
Another edge case here.
The [docs](http://docs.couchdb.org/en/1.6.1/query-server/javascript.html#require) only says
> The path should not start with a slash.

Do we need to show more comprehensive error message or this exception is our flaw?

```python
In [42]: ddoc
Out[42]: {'foo': {'bar': '42'}}

In [43]: pprint(resolve_module('./foo/bar'.split('/'), {}, root=ddoc))
---------------------------------------------------------------------------
Error                                     Traceback (most recent call last)
<ipython-input-43-e1de508a7691> in <module>()
----> 1 pprint(resolve_module('./foo/bar'.split('/'), {}, root=ddoc))

/usr/home/iblis/git/couchdb-python/couchdb/server/compiler.py in resolve_module(names, mod, root)
    146         if parent is None:
    147             raise Error('invalid_require_path',
--> 148                         'Object %r has no parent.' % idx + helper())
    149         return resolve_module(names, {
    150             'id': idx,

Error: ('invalid_require_path', "Object None has no parent.\n    id: None\n    names: ['foo', 'bar']\n    parent: None\n    current: None\n    root: {'foo': {'bar': '42'}}")
```